### ISSUE TITLE
Very laggy when starting 

### ISSUE THREAD

**Issue by stbth01** [Microsoft/vscode#63852]
We have written the needed data into your clipboard because it was too large to send. Please paste.

**Comment by stbth01** [comment]
Its hard to pin down, I do notice a spike in CPU usage so maybe its related
to #63314 but that ticket is not super helpful. I notice Omnisharp at the
top of the list particularly when I open a dotnet core project.

Sorry I know this not a lot of info but that is what is frustrating about
this is I don't know what is going on

**Comment by rebornix** [comment]
> We have written the needed data into your clipboard because it was too large to send. Please paste.

The information we ask is what the Issue Reporter put into the clipboard. You can also follow https://github.com/Microsoft/vscode/wiki/Performance-Issues to provide more detailed info.

**Comment by stbth01** [comment]
Issue Type: <b>Performance Issue</b>

adf

VS Code version: Code 1.29.1 (bc24f98b5f70467bc689abf41cc5550ca637088e, 2018-11-15T19:13:36.375Z)
OS version: Windows_NT x64 10.0.14393

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-4600U CPU @ 2.10GHz (4 x 2694)|
|GPU Status|2d_canvas: enabled<br>checker_imaging: disabled_off<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>rasterization: enabled<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Memory (System)|15.88GB (8.19GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|

</details><details>
<summary>Process Info</summary>

```
CPU %	Mem MB	   PID	Process
    3	    96	 11224	code main
    0	   207	   772	   window (profile-view.component.ts - mpr-lane1 - Visual Studio Code)
    0	   117	  4180	     extensionHost
    0	    33	  2880	       electron_node server.js 
    0	   237	  3356	       electron_node server.js 
    0	    79	  6312	       electron_node tslintServer.js 
    0	    34	 11904	       C:\Users\bhunscher.MATHEMATICA\.vscode\extensions\ms-vsliveshare.vsliveshare-0.3.1013\dotnet_modules\vsls-agent.exe --autoexit --pipe 311a70ccff1e4c608076662cffdf1a86 --service https://insiders.liveshare.vsengsaas.visualstudio.com/
    0	     9	  3648	         console-window-host (Windows internal process)
    0	   184	 13516	       electron_node tsserver.js 
    0	    45	 11792	         electron_node typingsInstaller.js typesMap.js 
    0	    38	 16136	       "C:\Users\bhunscher.MATHEMATICA\AppData\Local\Programs\Microsoft VS Code\Code.exe" "c:\Users\bhunscher.MATHEMATICA\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\json-language-features\server\dist\jsonServerMain" --node-ipc --clientProcessId=4180
    0	     6	  6768	     winpty-process
    0	     5	  5004	       "C:\Program Files\Git\bin\bash.exe"
    0	    10	 16472	         "C:\Program Files\Git\bin\..\usr\bin\bash.exe"
    0	     9	 10364	       console-window-host (Windows internal process)
    0	    35	 10304	     searchService
    0	    14	 13904	     watcherService 
    0	     9	  4520	       console-window-host (Windows internal process)
    0	   185	  8748	   window (package.json - Task2_DQ - Visual Studio Code)
    0	   110	 11572	     extensionHost
    0	    42	  4928	       "C:\Users\bhunscher.MATHEMATICA\AppData\Local\Programs\Microsoft VS Code\Code.exe" "c:\Users\bhunscher.MATHEMATICA\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\json-language-features\server\dist\jsonServerMain" --node-ipc --clientProcessId=11572
    0	     3	 14544	       cmd /s /c "C:\Users\bhunscher.MATHEMATICA\.vscode\extensions\ms-vscode.csharp-1.17.1\.omnisharp\1.32.8\OmniSharp.exe -s c:\Users\bhunscher.MATHEMATICA\Desktop\ProjectWork\Task2_DQ\MACBIS_DQ.sln --hostPID 11572 --stdio DotNet:enablePackageRestore=false --encoding utf-8 --loglevel information --plugin C:\Users\bhunscher.MATHEMATICA\.vscode\extensions\ms-vscode.csharp-1.17.1\.razor\OmniSharpPlugin\Microsoft.AspNetCore.Razor.OmniSharpPlugin.dll formattingOptions:useTabs=false formattingOptions:tabSize=4 formattingOptions:indentationSize=4"
    0	   168	 13588	         C:\Users\bhunscher.MATHEMATICA\.vscode\extensions\ms-vscode.csharp-1.17.1\.omnisharp\1.32.8\OmniSharp.exe  -s c:\Users\bhunscher.MATHEMATICA\Desktop\ProjectWork\Task2_DQ\MACBIS_DQ.sln --hostPID 11572 --stdio DotNet:enablePackageRestore=false --encoding utf-8 --loglevel information --plugin C:\Users\bhunscher.MATHEMATICA\.vscode\extensions\ms-vscode.csharp-1.17.1\.razor\OmniSharpPlugin\Microsoft.AspNetCore.Razor.OmniSharpPlugin.dll formattingOptions:useTabs=false formattingOptions:tabSize=4 formattingOptions:indentationSize=4
    0	     6	 14716	         console-window-host (Windows internal process)
    0	    34	 15032	       C:\Users\bhunscher.MATHEMATICA\.vscode\extensions\ms-vsliveshare.vsliveshare-0.3.1013\dotnet_modules\vsls-agent.exe --autoexit --pipe 61f6ecad9d5a43c192815242ed8c2012 --service https://insiders.liveshare.vsengsaas.visualstudio.com/
    0	     9	 19060	         console-window-host (Windows internal process)
    0	    34	 16712	       electron_node server.js 
    0	    44	 19124	       C:\Users\bhunscher.MATHEMATICA\.vscode\extensions\ms-vscode.csharp-1.17.1\.razor\rzls.exe -lsp --logLevel None
    0	     9	 13852	         console-window-host (Windows internal process)
    0	    14	 16948	     watcherService 
    0	     9	 15768	       console-window-host (Windows internal process)
    0	    35	 17120	     searchService
    0	     6	 19284	     winpty-process
    0	     8	  4188	       console-window-host (Windows internal process)
    0	     5	 12156	       "C:\Program Files\Git\bin\bash.exe"
    0	    10	  9904	         "C:\Program Files\Git\bin\..\usr\bin\bash.exe"
    0	    76	 13248	   shared-process
    0	   205	 15380	   gpu-process
    5	    72	 16904	   window (Issue Reporter)
```

</details>
<details>
<summary>Workspace Info</summary>

```
|  Window (profile-view.component.ts - mpr-lane1 - Visual Studio Code)
|  Window (package.json - Task2_DQ - Visual Studio Code)
|    Folder (mpr-lane1): 277 files
|      File types: ts(87) html(69) scss(38) svg(21) ttf(19) png(10) json(10)
|                  js(5) txt(4) md(2)
|      Conf files: tslint.json(2) package.json(1) tsconfig.json(1)
|    Folder (Task2_DQ): 2607 files
|      File types: cs(667) dll(423) ts(318) html(279) xml(168) js(114) css(80)
|                  svg(58) map(50) pp(48)
|      Conf files: csproj(4) webpack.config.js(2) package.json(2)
|                  tsconfig.json(2) tslint.json(2) sln(1) gulp.js(1)
|                  settings.json(1);
```

</details>
<details><summary>Extensions (25)</summary>

Extension|Author (truncated)|Version
---|---|---
ng-template|Ang|0.1.11
path-intellisense|chr|1.4.2
bracket-pair-colorizer|Coe|1.0.61
vscode-eslint|dba|1.7.0
xml|Dot|2.3.2
gitlens|eam|9.1.0
tslint|eg2|1.0.40
prettier-vscode|esb|1.7.2
Python-autopep8|him|1.0.2
beautify|Hoo|1.4.7
angular2-switcher|inf|0.2.0
csharpextensions|jch|1.3.0
vscode-nuget-package-manager|jmr|1.1.6
language-haskell|jus|2.5.0
Angular-BeastCode|Mik|7.0.20
prettify-json|moh|0.0.3
mssql|ms-|1.4.0
python|ms-|2018.11.0
csharp|ms-|1.17.1
vsliveshare|ms-|0.3.1013
debugger-for-chrome|msj|4.11.1
material-icon-theme|PKi|3.6.1
LiveServer|rit|5.3.1
aspnet-helper|sch|0.6.4
autoimport|ste|1.5.3


</details>

**Comment by stbth01** [comment]
I'm noticing rg.exe take up a lot of cpu when I open one project in particular but not others. Thanks for your help!

**Comment by roblourens** [comment]
Can you try downloading the Insiders' build and tell me whether it happens there? https://code.visualstudio.com/insiders

**Comment by stbth01** [comment]
It seems much better with the insider build but I went to add some of the
extensions I use and it seems the extensions are the cause of the slowness
at start up even with the insider. I see the elctron_node server.js process
taking up a lot of cpu, so I started playing the extension more and I think
VS Live Share is the main culprit.  I'm going to try working today with
that extension disabled.

I really like this extension so can I open issues for that extension or
give more precise information?

Thanks for your help
